python/examples/querying-a-single-package.py

```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/python
+
+# Query a single installed package by its name.
+# A Python equivalent for `rpm -q hello`
+
+import rpm
+
+name = "hello"
+ts = rpm.TransactionSet()
+mi = ts.dbMatch("name", name)
+hdr = next(mi, None)
+if hdr:
```

Comment:
This should be in a loop just like in your querying-all-installed version - there can always be more than one package with a given name, and callers better handle that correctly.

Reply:
Looping over the return value (`for hdr in mi:`) also has the advantage that you can skip the `next()` call and the `if hdr` check; if an empty list is returned it will simply cause the loop the be skipped.

Reply:
(You can also check for an empty list after the loop, e.g. this will print "No results" only when the list is empty):

```python
mi = ts.dbMatch("name", name)
for hdr in mi:
    ...
if not mi:
    print("No results")
```

Reply:
Actually, I take that back. Because `mi` is an iterator, not a list, it will become `False` when exhausted, so that code will always print "No results" at the end of the loop. My mistake.